view/src/components/foodbanks.js

```diff
@@ -0,0 +1,937 @@
+ * where * represents a digit
+ */
+function TextMaskCustom(props) {
+  const { inputRef, ...other } = props;
+
+  return (
+    <MaskedInput
+      {...other}
+      ref={(ref) => {
+        inputRef(ref ? ref.inputElement : null);
+      }}
+      mask={[
+        "(",
+        /[1-9]/,
+        /\d/,
+        /\d/,
+        ")",
+        " ",
+        /\d/,
+        /\d/,
+        /\d/,
+        "-",
+        /\d/,
+        /\d/,
+        /\d/,
+        /\d/,
+      ]}
+      placeholderChar={"\u2000"}
+      showMask
+    />
+  );
+}
+
+TextMaskCustom.propTypes = {
+  inputRef: PropTypes.func.isRequired,
+};
+
+/**
+ * This class represents a Food Bank component, which is a sub-page of the
+ * home page where food bank objects are visualized, created, updated, edited,
+ * and deleted.
+ */
+class Foodbank extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      // food bank states
+      foodbanks: "",
+      foodbankName: "",
+      location: "",
+      locationId: "",
+      hours: "",
+      foodbankId: "",
+      contactPhone: "(1  )    -    ",
+      contactName: "",
+      contactEmail: "",
+      forklift: false,
+      pallet: false,
+      loadingDock: false,
+      maxLoadSize: "",
+      refrigerationSpaceAvailable: "",
+      foodbankTags: [],
+      // page states
+      errors: [],
+      open: false,
+      uiLoading: true,
+      buttonType: "",
+      viewOpen: false,
+    };
+
+    this.handleDelete = this.handleDelete.bind(this);
+    this.handleEditClick = this.handleEditClick.bind(this);
+    this.handleViewOpen = this.handleViewOpen.bind(this);
+  }
+
+  /**
+   * Given an event, this function updates a state (the target of the event)
+   * with a new value
+   * @param event The event that is attempting to update a state
+   */
+  handleChange = (event) => {
+    this.setState({
+      [event.target.name]: event.target.value,
+    });
+  };
+
+  /** Used to uncheck and check the form checkboxes */
+  handleChecked = (name) => (event) => {
+    this.setState({ ...this.state, [name]: event.target.checked });
+  };
+
+  /** Used to update tags in form */
+  onTagsChange = (event, values) => {
+    this.setState({
+      // food bank state
+      farmTags: values,
+    });
+  };
+
+  /** Used to update location from address autocomplete component */
+  handleLocation = (newValue) => {
+    if (newValue === null) {
+      return;
+    }
+    this.setState({
+      location: newValue.description,
+      locationId: newValue.place_id,
+    });
+  };
+
+  /** Returns the authentication token stored in local storage */
+  getAuth = () => {
+    authMiddleWare(this.props.history);
+    return localStorage.getItem("AuthToken");
+  };
+
+  /** Load in all of the current todos when the component has mounted */
+  componentDidMount() {
+    axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
+    axios
+      .get("/foodbanks")
+      .then((response) => {
```

Comment:
No need to change this unless you want to, but because you're using Babel I'm pretty sure you can use async and await keywords in this code.  Assuming `axios.get()` is returning a Promise (and not just exposing an API that looks like one in order to confuse me), you could rewrite this as:

```
async componentDidMount() {
  axios.defaults.headers.common = { Authorization: `${this.getAuth()}` };
  try {
    const response = await axios.get("/foodbanks");
    this.setState({
      foodbanks: response.data,
      uiLoading: false,
    });
  } catch (err) {
    console.log(err);
  }
  // ...
}
```

It basically lets you pretend that `axios.get()` returns a Response instead of a Promise that resolves to a Response.

Reply:
Oh that is so cool! I love the ability in JS and React to move between approaches so easily. I did a little research to ensure that you can include async and await in React lifestyle methods (such as componentDidMount) and it seems totally fine. I am gonna bring this up on the next pod call to see if my pod mates think switching to this style would make the code easier to work with. As always, thanks for the great tip!